lib/bolt/config.rb

```diff
@@ -224,14 +237,21 @@ def update_from_file(data)
       @trusted_external = if data.key?('trusted-external-command')
                             File.expand_path(data['trusted-external-command'], @boltdir.path)
                           end
+
+      if data.key?('apply_settings')
+        @apply_settings, extra_opts = data['apply_settings'].partition do |k, _|
+          APPLY_SETTINGS.keys.include?(k)
+        end.map(&:to_h)
+        unless extra_opts.empty?
+          @logger.warn("Apply settings #{extra_opts.keys.join(', ')} are not valid in Bolt")
```

Comment:
This warning will never get displayed since the logger isn't configured until after the config.

Reply:
For the `puppetfile` section, we just ignore any unknown settings. I'm not sure whether that's the right thing or not, but we should probably be consistent between the two of them, whether the behavior is silence, warning, or error.

Reply:
I mostly added it because I imagine we'll be changing which settings are supported relatively frequently in the next few weeks, and thought people might just try to add all their preferred Puppet settings without checking which are supported and which aren't. I don't know of any settings that would have harmful effects if they weren't set as intended though, it might be easiest to remove this and add it if it's a problem.

Reply:
With the exception of noop, which is already supported.

Reply:
I think _most_ Puppet settings will either have no effect or have a harmful effect. So I do like the idea of not just supporting everything.